File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        m, r = 1, [False]*len(nums)
        for index, item in enumerate(nums):
            #print(index, item)
            m = max(m-1, item)
            r[index] = True
            if(m == 0): break
        return r[len(nums)-1] == True
            
        

File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let max = 0;
    for(let i = 0 ; i < nums.length ; i++){
        if(max<i) return false;
        max = Math.max(max, nums[i] + i);
    }
    return true;
};